### PR TITLE
Set parse_options from fuzzdata and invoke to_string()

### PR DESCRIPTION
**Pull request by pauldreik**
This should cover more of the code. I have run this locally, no crashes found.

### CODE REVIEW

File: fuzzing/fuzz_parser.cpp [hunk 1]
```diff
@@ -39,14 +40,27 @@ fuzz_parser(string_view sv)
...
+        const uint8_t* data, size_t size)
 {
+    if(size<1)
+        return 0;
+
+    parse_options popt;
+    popt.allow_comments=!!(data[0]&0x1);
+    popt.allow_trailing_commas=!!(data[0]&0x2);
+    popt.allow_invalid_utf8=!!(data[0]&0x4);
+    popt.max_depth= (data[0]>>3);
```
Comment (vinniefalco) [inline fuzzing/fuzz_parser.cpp:53]: hey that's really cool!!!

Comment (pauldreik) [reply fuzzing/fuzz_parser.cpp:53]: Yep, only cool kids do bit shifts and stuff :-)

Comment (vinniefalco) [reply fuzzing/fuzz_parser.cpp:53]: LOL no... I mean, the mechanism for prodding the fuzzing tool to fill out the parse options

Comment (pauldreik) [reply fuzzing/fuzz_parser.cpp:53]: Why are you not impressed with my coding skillz? I can do bitwise or, too!

On a more serious note, yes it is awesome to fill out input data from the fuzzer stream. Have a look at the curl fuzzer, which sets flags and then cuts different parts of the input data as the (fake) server response.

**Pull request closed without merging**